src/Microsoft.IdentityModel.Tokens/Validation/Validators.Audience.cs

```diff
@@ -102,7 +102,7 @@ internal static AudienceValidationResult ValidateAudience(IList<string> audience
         {
             string? validAudience = null;
 
-            List<string> validAudiences = validationParameters.ValidAudiences.ToList();
+            List<string> validAudiences = [.. validationParameters.ValidAudiences];
```

Comment:
Is this not creating a new list and appending the values of `ValidAudiences`? I need someone with more C# expertise to chime in here.
In my code I used casting in the way of:

```csharp
if (validationParameters.ValidAudiences is not List<string> validAudiences)
    validAudiences = validationParameters.ValidAudiences.toList();
```

And then work with `validAudiences` instead of `validationParameters.ValidAudiences`.

This way in the most common scenario where the assigned value is indeed a `List<string>` this cast does nothing other than assigning to the new variable. And if for some reason the input happens to not be a `List<string>`, it will be converted.

Reply:
We should not be copying the audiences.
From the first call in the call graph, we should be using IList.